crates/core/tedge_mapper/src/core/mapper.rs

```diff
@@ -121,12 +121,45 @@ impl Mapper {
             self.process_message(message).await;
         }
 
-        // Continue processing messages after the sync period
-        while let Some(message) = self.input.next().await {
-            self.process_message(message).await;
+        // Create inotify steam for capturing the inotify events.
+        let mut inotify = create_inotify_watch(ops_dir.clone()).expect("No operations directory");
+        let mut buffer = [0; 1024];
+        let mut stream = inotify
+            .event_stream(&mut buffer)
+            .expect("Failed to create the inotify stream");
+
+        loop {
+            tokio::select! {
+                msg =  self.input.next() => {
+                            match msg {
+                                Some(message)=> {self.process_message(message).await;}
+                                None => {error!("Failed to extract the mqtt message");}
+
+                        }
+                    }
+                event = stream.next() => {
+                    match event {
+                        Some(ev) => {
+                                match  process_inotify_events(ops_dir.clone(), ev) {
+                                    Ok(discovered_ops) => {
+                                        let operations =  self.converter.process_operation_update_message(discovered_ops).expect("Could not process operation update request");
```

Comment:
`expect` panics on error as `unwrap() does.

Please have a look to what has been done for the other methods of the converter trait. The `process_message()` method *never* fails. This is done by it invoking the `try_process_message()` and transforming any error into an MQTT message to be published on the `tedge/errors` topics.